Utilities/CodeGeneration/EdFi.Ods.CodeGen/Generators/Resources/ResourcePropertyRenderer.cs

```diff
@@ -120,7 +120,7 @@ public object AssembleProperties(ResourceClassBase resource)
                         props.Add(
                             property.IsPersonOrUsi() && !templateContext.IsExtension
                                 ? PropertyData.CreateUsiPrimaryKey(property)
-                                : property.HasAssociations() && !property.IsDirectDescriptorUsage
+                                : property.HasAssociations() && property.EntityProperty.IncomingAssociations.Any(a => a.IsRequired) && !property.IsDirectDescriptorUsage
```

Comment:
This is the important part. If none of the associations is required, I render a Standard property instead of a Referenced one. All the associations that reference ChartOfAccounts.FiscalYear are optional.

Without this, CodeGen renders an implicit property using the first association (BalanceSheetDimension) [here](https://github.com/Ed-Fi-Alliance-OSS/Ed-Fi-ODS/blob/main/Utilities/CodeGeneration/EdFi.Ods.CodeGen/Generators/Resources/PropertyData.cs#L140), and it fails during post/put if no BalanceSheetDimensionReference is defined.